sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ReadFromKafkaDoFn.java

```diff
@@ -557,20 +558,16 @@ public ProcessContinuation processElement(
   }
 
   private boolean topicPartitionExists(
-      TopicPartition topicPartition, Map<String, List<PartitionInfo>> topicListMap) {
+      TopicPartition topicPartition, List<PartitionInfo> partitionInfos) {
     // Check if the current TopicPartition still exists.
     Set<TopicPartition> existingTopicPartitions = new HashSet<>();
-    for (List<PartitionInfo> topicPartitionList : topicListMap.values()) {
-      topicPartitionList.forEach(
-          partitionInfo -> {
-            existingTopicPartitions.add(
-                new TopicPartition(partitionInfo.topic(), partitionInfo.partition()));
-          });
-    }
-    if (!existingTopicPartitions.contains(topicPartition)) {
-      return false;
-    }
-    return true;
+    partitionInfos.forEach(
+        partitionInfo -> {
+          existingTopicPartitions.add(
+              new TopicPartition(partitionInfo.topic(), partitionInfo.partition()));
+        });
+
+    return existingTopicPartitions.contains(topicPartition);
   }
```

Comment:
The whole list needs to be iterated to construct the set, so this might as well just be `partitionInfos.stream().anyMatch(partitionInfo -> partitionInfo.partition() == topicPartition.partition())`, right?

Reply:
Good call, I'll make that change